packages/@aws-cdk/aws-servicecatalogappregistry/lib/application-associator.ts

```diff
@@ -17,58 +17,86 @@ export interface ApplicationAssociatorProps {
  * needs to be associated separately.
  *
+ * If cross region stack is detected, then this construct will give a warning and skip that association.
+ * Only stacks and an application within the same region will be associated.
+ *
  * If cross account stack is detected, then this construct will automatically share the application to consumer accounts.
  * Cross account feature will only work for non environment agnostic stacks.
  */
 export class ApplicationAssociator extends Construct {
   /**
-   * Created or imported application.
+   * Map of region to application. An application can be one that is created or imported.
+   * Because a stack can only be associated with one application in a region,
+   * one region will map to only one application.
    */
-  private readonly application: IApplication;
+  private readonly regionToApplication: Map<string, IApplication> = new Map();
   private readonly associatedStages: Set<cdk.Stage> = new Set();
 
   constructor(scope: cdk.App, id: string, props: ApplicationAssociatorProps) {
     super(scope, id);
 
-    if (props.applications.length != 1) {
-      throw new Error('Please pass exactly 1 instance of TargetApplication.createApplicationStack() or TargetApplication.existingApplicationFromArn() into the "applications" property');
+    if (props.applications.length == 0) {
+      throw new Error('Please pass at least 1 instance of TargetApplication.createApplicationStack() or TargetApplication.existingApplicationFromArn() into the "applications" property');
     }
 
-    const targetApplication = props.applications[0];
-    this.application = targetApplication.bind(scope).application;
-    cdk.Aspects.of(scope).add(new CheckedStageStackAssociator(this));
+    props.applications.forEach((targetApplication) => {
+      const application = targetApplication.bind(scope).application;
+      const arnComponent = cdk.Arn.split(application.applicationArn, cdk.ArnFormat.SLASH_RESOURCE_SLASH_RESOURCE_NAME);
+      const region = arnComponent.region!;
+      this.regionToApplication.set(region, application);
+      cdk.Aspects.of(scope).add(new CheckedStageStackAssociator(this, region));
+    });
   }
 
   /**
-   * Associate this application with the given stage.
+   * Associate an application with the given stage of the same region.
+   * If there does not exist an application for the given region, then skip the
+   * association.
    *
    */
   public associateStage(stage: cdk.Stage): cdk.Stage {
     this.associatedStages.add(stage);
-    cdk.Aspects.of(stage).add(new CheckedStageStackAssociator(this));
+    const application = this.regionToApplication.get(stage.region!);
+    if (application) {
+      cdk.Aspects.of(stage).add(new CheckedStageStackAssociator(this, stage.region!));
+    } else {
+      cdk.Annotations.of(stage).addWarning(`There is no application defined in ${stage.region}. Skipping association.`);
+    }
```

Comment:
Definitely an error

Reply:
I'm checking with our product manager on the customer's expected behavior but I've gone ahead with changing this to error for now.